packages/docs-infra/src/CodeHighlighter/CodeErrorHandler.tsx

```diff
@@ -0,0 +1,23 @@
+'use client';
+
+import * as React from 'react';
+import { useErrors } from '../useErrors';
+import { ErrorHandlerProps } from './types';
+
+export function CodeErrorHandler({ errors }: ErrorHandlerProps) {
+  const context = useErrors();
+  errors = context.errors || errors;
+
+  if (!errors || errors.length === 0) {
+    return <div>An error occurred, but details were not provided.</div>;
```

Comment:
Is this ever going to be shown to end-users? What's the use-case of this? Feels like for end-users, this should just silently fail and simply log an error (or report to a remote log, but let's not get ahead of ourselves).

Reply:
Yes, this is the default error shown when no error context is provided on the client. The Error component has to be rendered before crossing a client boundary, so we have to update its props using context if there is an error on the client.